qabelbox/src/main/java/de/qabel/android/config/AndroidPersistence.java

```diff
@@ -0,0 +1,187 @@
+            return false;
+        }
+
+        ContentValues contentValues = new ContentValues();
+        contentValues.put(STR_BLOB, serialize(object.getPersistenceID(), object));
+
+        String[] whereArgs = { object.getPersistenceID() };
+        return database.update(getTableNameForClass(object.getClass()),
+                contentValues, STR_ID_QUERY, whereArgs) != -1;
+    }
+
+    @Override
+    public boolean updateOrPersistEntity(Persistable object) {
+        if (getEntity(object.getPersistenceID(), object.getClass()) == null) {
+            return persistEntity(object);
+        }
+        else {
+            return updateEntity(object);
+        }
+    }
+
+    @Override
+    public boolean removeEntity(String id, Class cls) {
+        if (id == null || cls == null) {
+            throw new IllegalArgumentException("Arguments cannot be null!");
+        }
+        if (id.isEmpty()) {
+            throw new IllegalArgumentException("ID cannot be empty!");
+        }
+
+        String[] whereArgs = { id };
+        return database.delete(getTableNameForClass(cls), STR_ID_QUERY, whereArgs) == 1;
+    }
+
+    @Override
+    public <U extends Persistable> U  getEntity(String id, Class<? extends U> cls) {
+        if (id == null || cls == null) {
+            throw new IllegalArgumentException("Arguments cannot be null!");
+        }
+        if (id.isEmpty()) {
+            throw new IllegalArgumentException("ID cannot be empty!");
+        }
+
+        String[] columns = {STR_BLOB};
+        String[] selectionArgs = { id };
+
+        Cursor cursor = null;
+        try {
+            cursor = database.query(getTableNameForClass(cls), columns, STR_ID_QUERY,
+                    selectionArgs, null, null, null);
+
+            if (cursor.moveToFirst()) {
+                return (U) deserialize(id, cursor.getBlob(0));
+            }
+        } catch (SQLiteException e) {
+            LOGGER.debug("Couldn't get entity! " + e.getLocalizedMessage());
+        } finally {
+            if (cursor != null) {
+                cursor.close();
+            }
+        }
+        return null;
+    }
+
+    @Override
+    public <U extends Persistable> List<U> getEntities(Class<? extends U> cls) {
+        if (cls == null) {
+            throw new IllegalArgumentException("Arguments cannot be null!");
+        }
+        List<U> objects = new ArrayList<>();
+
+        String[] columns = {STR_ID, STR_BLOB};
+
+        Cursor cursor = null;
+        try {
+            cursor = database.query(getTableNameForClass(cls), columns,
+                    null, null, null, null, null);
+            if (cursor.moveToFirst()) {
+                do {
+                    objects.add((U) deserialize(cursor.getString(0),
+                            cursor.getBlob(1)));
+                } while (cursor.moveToNext());
+            }
+        }
+        catch (SQLiteException e){
+            LOGGER.info("Table does not exist!");
```

Comment:
would be more consistent with `getEntity` to return `null` on error instead of an empty `ArrayList`, wouldn't it?